test/gtest/uct/uct_test.cc

```diff
@@ -470,20 +486,35 @@ bool uct_test::check_atomics(uint64_t required_ops, atomic_mode mode) {
 
 void uct_test::modify_config(const std::string& name, const std::string& value,
                              bool optional) {
-    ucs_status_t status;
+    ucs_status_t status = UCS_OK;
+
+    if (m_cm_config != NULL) {
+        status = uct_config_modify(m_cm_config, name.c_str(), value.c_str());
+        if (status == UCS_OK) {
+            return;
+        } else if (status != UCS_ERR_NO_ELEM) {
+            UCS_TEST_ABORT("Couldn't modify cm config parameter: " << name.c_str() <<
+                           " to " << value.c_str() << ": " << ucs_status_string(status));
+        }
+    }
 
-    status = uct_config_modify(m_iface_config, name.c_str(), value.c_str());
-    if (status == UCS_ERR_NO_ELEM) {
-        status = uct_config_modify(m_md_config, name.c_str(), value.c_str());
-        if (status == UCS_ERR_NO_ELEM) {
-            test_base::modify_config(name, value, optional);
-        } else if (status != UCS_OK) {
-            UCS_TEST_ABORT("Couldn't modify pd config parameter: " << name.c_str() <<
+    if (m_iface_config != NULL) {
+        status = uct_config_modify(m_iface_config, name.c_str(), value.c_str());
+        if (status == UCS_OK) {
+            return;
+        } else if (status != UCS_ERR_NO_ELEM) {
+            UCS_TEST_ABORT("Couldn't modify iface config parameter: " << name.c_str() <<
                            " to " << value.c_str() << ": " << ucs_status_string(status));
         }
+    }
+
+    ASSERT_TRUE(status == UCS_ERR_NO_ELEM);
```

Comment:
one of configs, m_cm_config or m_iface_config, must be non-NULL? right?

Reply:
right

Reply:
i think this assert is not needed. it's clear that if status was NO_ELEM we didnt get here
and in any case it should have been ucs_assert() which indicates programming assert and not ASSERT_TRUE which is a gtest macro